core/src/devicemanager.cpp

```diff
@@ -40,16 +40,25 @@ void DeviceManager::addDevice(Device *d)
 	QString id = d->id();
 	map[id] = d;
 	di->setParent(this);
+	QList<Plugin *> availablePlugins = di->plugins();
+	if(!m_plugins.isEmpty()) {
+		for(Plugin *p : qAsConst(availablePlugins)) {
+			if(!m_plugins.contains(p->name())) {
+				p->setEnabled(false);
+			}
+		}
+	}
```

Comment:
You can do this steps inside the api without changing the addDevice method.

Reply:
steps removed from addDevice